### PR TITLE
Comms arm

### LINKED ISSUE #503: Implement new comms system on Arm Teensy code
The new comms system should be implemented in the Arm teensy code. All the commands should be documented

### PR DESCRIPTION
**Pull request by MartensCedric**
# Assignee Section

## Description
Refactored the Arm code to be supported by the new comms systems

- Reorganized the directory structure
- Use the `internal_comms` library

### Steps for Testing
```
For each feature f on the arm:
   test(f)
```
closes #503 

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.

- [ ] Local Test Performed Successfully
- [ ] [General Integration Test](https://docs.google.com/document/d/1ug0CpA1cIzURP8DDFSvCt2CEJJSwJ6Ta6B1LG_hYk6I/edit) Performed Successfully

For Pull Requests that do not include code changes, it is not required to perform the tests above.

### CODE REVIEW

File: robot/rover/Arm/src/commands/ArmCommandCenter.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    case COMMAND_RESET_SINGLE_MOTOR:
+      resetSingleMotor(*rawArgs);
+      break;
+    case COMMAND_BUDGE_MOTORS:
+      uint8_t motorsToMove[3] = {2, 4, 5};
+      bool moveCW[3] = {true, false, true};
+      budgeMotors(motorsToMove, moveCW);
+      break;
+    case COMMAND_MOVE_MULTIPLE_MOTORS:
+      moveMultipleMotors(0, 0);
```
Comment (MartensCedric) [inline robot/rover/Arm/src/commands/ArmCommandCenter.cpp:94]: some of these values are 0, or 1.0, this is just placeholder until my next commit

File: robot/rover/Arm/include/commands/ArmCommandCenter.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
+//
+// Created by cedric on 2021-01-23.
+//
+
+#ifndef ARM_ARMCOMMANDCENTER_H
+#define ARM_ARMCOMMANDCENTER_H
+
+#include "../../../internal_comms/include/CommandCenter.h"
```
Comment (VimFreeman) [inline robot/rover/Arm/include/commands/ArmCommandCenter.h:8]: Can't you include it the same way as in PDSCommandCenter.h in PR #520?

File: robot/rover/Arm/src/PinSetup.cpp [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #include <Arduino.h>
-#include "PinSetup.h"
+#include "../include/PinSetup.h"
```
Comment (VimFreeman) [inline robot/rover/Arm/src/PinSetup.cpp:2]: Same thing here, I think CMake takes care of the include paths..

File: robot/rospackages/src/mcu_control/scripts/ArmNode.py [hunk 4]
```diff
@@ -4,101 +4,86 @@
...
+    return None
+
+# Pin definitions
+arm_pin = 11
+wheel_pin = 13
+science_pin = 15
+
+teensy_pins = [arm_pin, wheel_pin, science_pin]
+
+ser = serial.Serial('/dev/ttyACM0', 57600) # you sure this is good for the jetson tim?
```
Comment (VimFreeman) [inline robot/rospackages/src/mcu_control/scripts/ArmNode.py:40]: we're gonna need a separate node that handles traffic cause this way there are gonna be collisions. 
For testing it's OK as long as you only have this one node running

File: robot/rospackages/src/mcu_control/scripts/ArmNode.py [hunk 5]
```diff
@@ -113,58 +98,148 @@ def stripFeedback(data):
...
+#    except rospy.ROSInterruptException:
+#        pass
+
+#    def shutdown_hook():
+#        rospy.logwarn('This node ('+node_name+') is shutting down')
+#        ser.close() # good practice to close the serial port
+#        # do I need to clear the serial buffer too?
+#        time.sleep(1) # give ROS time to deal with the node closing (rosbridge especially)
+
+#    rospy.on_shutdown(shutdown_hook)
```
Comment (VimFreeman) [inline robot/rospackages/src/mcu_control/scripts/ArmNode.py:245]: saving in case it's needed?

**Comment by MartensCedric** [comment]
@VimFreeman pls :eye:

File: robot/rover/Arm/include/Vsense.h [hunk 6]
```diff
@@ -12,14 +12,14 @@ void vbatt_read() {
...
+  //UART_PORT.print("ARM battery voltage: ");
+  //UART_PORT.println(vbatt);
 
   if (vbatt < 12.0) {
-    UART_PORT.println("ASTRO WARNING! BATTERY VOLTAGE IS LOW! DISCONNECT IMMEDIATELY");
+    //UART_PORT.println("ASTRO WARNING! BATTERY VOLTAGE IS LOW! DISCONNECT IMMEDIATELY");
   }
   else if (vbatt > 16.8) {
-    UART_PORT.println("ASTRO WARNING! BATTERY VOLTAGE IS HIGH! DISCONNECT IMMEDIATELY!");
+    //UART_PORT.println("ASTRO WARNING! BATTERY VOLTAGE IS HIGH! DISCONNECT IMMEDIATELY!");
```
Comment (VimFreeman) [inline robot/rover/Arm/include/Vsense.h:22]: what are we gonna do with this

Comment (MartensCedric) [reply robot/rover/Arm/include/Vsense.h:22]: YEET

Comment (MartensCedric) [reply robot/rover/Arm/include/Vsense.h:22]: later

File: robot/rospackages/src/mcu_control/scripts/ArmNode.py (hunk 5 above)
Comment (MartensCedric) [reply robot/rospackages/src/mcu_control/scripts/ArmNode.py:255]: It will remind him that he has to send battery voltage, motor angles etc

**Review by VimFreeman — APPROVED** [review]
.

**Pull request merged**